### PR TITLE
[Improve][Connector-V2][Http]Improve json parse option rule for all http connector

### PR DESCRIPTION
**Pull request by TaoZex**
## Purpose of this pull request

Improve json parse option rule for all http connector

## Check list

* [ ] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/incubator-seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/incubator-seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/incubator-seatunnel/blob/dev/seatunnel-dist/pom.xml)

### CODE REVIEW

**Review by PraveenNanda124 — COMMENTED** [review]
Good work

File: docs/en/connector-v2/source/Gitlab.md [hunk 1]
```diff
@@ -26,6 +26,8 @@ Used to read data from Gitlab.
 | format                      | String | No       | json          |
 | params                      | Map    | No       | -             |
 | body                        | String | No       | -             |
+| json_field                  | Config | No       | -             |
```
Comment (EricJoy2048) [inline docs/en/connector-v2/source/Gitlab.md:29]: It seems `json_field` is `Object` type.

File: docs/en/connector-v2/source/Gitlab.md [hunk 2]
```diff
@@ -124,6 +130,140 @@ connector will generate data as the following:
...
+        price = string
+      }
+    }
+  }
+}
+```
+
+- Test data can be found at this link [mockserver-jsonpath-config.json](../../../../seatunnel-e2e/seatunnel-connector-v2-e2e/connector-http-e2e/src/test/resources/mockserver-jsonpath-config.json)
+- See this link for task configuration [http_jsonpath_to_assert.conf](../../../../seatunnel-e2e/seatunnel-connector-v2-e2e/connector-http-e2e/src/test/resources/http_jsonpath_to_assert.conf).
+
```
Comment (EricJoy2048) [inline docs/en/connector-v2/source/Gitlab.md:266]: Please add changed log for all connectors updated.

**Review by EricJoy2048 — COMMENTED** [review]
I see we need update all connectors base on http connector when the http connector add some parameters? Can we optimize the code？

File: docs/en/connector-v2/source/Gitlab.md (hunk 1 above)
Comment (TaoZex) [reply docs/en/connector-v2/source/Gitlab.md:29]: Please give us some advice. @liugddx

Comment (liugddx) [reply docs/en/connector-v2/source/Gitlab.md:29]: > 
> 
> Please give us some advice. @liugddx

It is `config` type.

Comment (TaoZex) [reply docs/en/connector-v2/source/Gitlab.md:29]: > > 
> > Please give us some advice. @liugddx
> 
> It is `config` type.

Thanks.

**Comment by TaoZex** [comment]
> I see we need update all connectors base on http connector when the http connector add some parameters? Can we optimize the code？


I have tried to optimize the code and do not need to make any code changes to other http connectors for the new http feature. Please review it again.

**Review by EricJoy2048 — DISMISSED** [review]

**Comment by EricJoy2048** [comment]
> > I see we need update all connectors base on http connector when the http connector add some parameters? Can we optimize the code？
> 
> I have tried to optimize the code and do not need to make any code changes to other http connectors for the new http feature. Please review it again.

Thanks.

**Comment by EricJoy2048** [comment]
@liugddx  PTAL

**Review by liugddx — APPROVED** [review]

**Review by EricJoy2048 — APPROVED** [review]

**Pull request merged**